docs/template_plugin/tests/functional/op_reference/broadcast.cpp

```diff
@@ -0,0 +1,528 @@
+    Exec();
+}
+
+struct BroadcastParamsTestHelper {
+    BroadcastParamsTestHelper(
+        const Shape& shapeA,
+        const Shape& shapeR,
+        const AxisSet& axes, const std::string& testcaseName = "") :
+        shapeA(shapeA), shapeR(shapeR),
+        axes(axes), testcaseName(testcaseName) {}
+
+    Shape shapeA;
+    Shape shapeR;
+    AxisSet axes;
+    std::string testcaseName;
+};
+
+class ReferenceBroadcastTestTestHelper : public testing::TestWithParam<BroadcastParamsTestHelper>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        std::vector<float> inpData(shape_size<const Shape>(params.shapeA));
+        iota(inpData.begin(), inpData.end(), 1.f);
+        const auto refA = CreateTensor(params.shapeA, element::f32, inpData);
+        inputData = {refA};
+    }
+
+    void SetUp1() {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        std::vector<float> inpData(shape_size<const Shape>(params.shapeA));
+        iota(inpData.begin(), inpData.end(), 1.f);
+        const auto wrkA = CreateTensor(params.shapeA, element::f32, inpData);
+        inputData = {wrkA};
+    }
+
+    static std::string getTestCaseName(const testing::TestParamInfo<BroadcastParamsTestHelper>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "aShape=" << param.shapeA;
+        result << "_rShape=" << param.shapeR;
+        if (param.testcaseName != "") {
+            result << "_axes=" << param.axes;
+            result << "_=" << param.testcaseName;
+        } else {
+            result << "_axes=" << param.axes;
+        }
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const BroadcastParamsTestHelper& params) {
+        const auto A = std::make_shared<opset1::Parameter>(element::f32, params.shapeA);
+        const auto shape_const = opset1::Constant::create(element::u64, Shape{params.shapeR.size()}, params.shapeR);
+        std::shared_ptr<Node> broadcast;
+        if (params.axes.size() > 0) {
+            auto axes_const = opset1::Constant::create(element::i64, Shape{params.axes.size()}, params.axes.to_vector());
+            broadcast = std::make_shared<opset1::Broadcast>(A, shape_const, axes_const);
+        } else {
+            broadcast = std::make_shared<opset1::Broadcast>(A, shape_const);
+        }
+        auto f = std::make_shared<Function>(broadcast, ParameterVector{A});
+        return f;
+    }
+
+protected:
+    void GenerateRefOutData() {
+        actualOutData.clear();
+        for (const auto &output : executableNetwork.outputs()) {
+            actualOutData.emplace_back(inferRequest.get_tensor(output));
+        }
+        refOutData = actualOutData;
+    }
+};
+
+TEST_P(ReferenceBroadcastTestTestHelper, CompareWithRefs) {
+    LoadNetwork();
```

Comment:
Why do we need a custom pipeline here?

Reply:
> Why do we need a custom pipeline here?

The broadcast backend test has `broadcast_test_helper`. It calls the test function two times. And the second function uses the output of the first function.
So this test needs output without validation. Do we have any methods in this kind of test?